File: src/yh_turtle/scripts/yh_turtle_circle.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
import rospy
from geometry_msgs.msg import Twist
from turtlesim.srv import TeleportAbsolute
from std_srvs.srv import Empty
import sys
import math

class TurtleCricle():
    
    def __init__(self, radius):
        self.pub = rospy.Publisher("/turtle1/cmd_vel", Twist, queue_size=100)
        self.client_Tele = rospy.ServiceProxy("/turtle1/teleport_absolute",TeleportAbsolute)
        self.client_clear = rospy.ServiceProxy("/clear", Empty)
        self.radius = radius
        
    def run(self):
        
        self.client_Tele(5.44445, 5.44445 - self.radius ,0)
        self.client_clear()
        
        msg = Twist()
        
        msg.linear.x = self.radius
        msg.angular.z = 1
        
        looprate = rospy.Rate(1)
        
        while not rospy.is_shutdown():
            self.pub.publish(msg)

#def talker():
    #rospy.init_node("yh_turtle_circle")
    #pub = rospy.Publisher("/turtle1/cmd_vel", Twist, queue_size=100)
    
    #msg = Twist()
    
    #msg.linear.x = int(sys.argv[1])
    #msg.angular.z = int(sys.argv[2])
    
    #a = rospy.Rate(1)
    
    #while not rospy.is_shutdown():
    
        #pub.publish(msg)
        #a.sleep()
    
if __name__ == "__main__":
    
    try:
        
        rospy.init_node("yh_turtle_circle2")
        radius = float(sys.argv[1])
        turtle_circle = TurtleCricle(radius)
        turtle_circle.run()
        
    except rospy.ROSInterruptException:
        pass

File: src/yh_turtle/src/yh_turtle_circle.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include "turtlesim/TeleportAbsolute.h"
#include "std_srvs/Empty.h"
#include <cstdlib>


class TurtleCricle
{
    private:

        ros::NodeHandle nh;
        ros::Publisher pub;
        ros::ServiceClient client_clear;
        ros::ServiceClient client_tele;
        double radius;

    public:

        TurtleCricle(double r)
        {
            pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 100);
            client_clear = nh.serviceClient<std_srvs::Empty>("/clear");
            client_tele = nh.serviceClient<turtlesim::TeleportAbsolute>("/turtle1/teleport_absolute");
            radius = r;
        }
        
        void run(void)
        {
            turtlesim::TeleportAbsolute srv_tele;
            srv_tele.request.x = 5.544445;
            srv_tele.request.y = 5.544445 - radius;
            srv_tele.request.theta = 0;

            std_srvs::Empty clear;
            client_tele.call(srv_tele);
            client_clear.call(clear);

            ros::Rate loop_rate(1);
            geometry_msgs::Twist msg;

            msg.linear.x = radius * 3;
            msg.angular.z = 3.;

            while (ros::ok())
            {
                /* code */
                pub.publish(msg);
                loop_rate.sleep();
            }
            

        }
};

int main(int argc, char **argv)
{
    ros::init(argc,argv,"yh_turtle_circle");
    double radius = atof(argv[1]);
    TurtleCricle tt(radius);
    tt.run();

    return 0;
}